ios/REANodesManager.m

```diff
@@ -174,12 +196,36 @@ - (void)performOperations
   if (_operationsInBatch.count != 0) {
     NSMutableArray<REANativeAnimationOp> *copiedOperationsQueue = _operationsInBatch;
     _operationsInBatch = [NSMutableArray new];
+    
+    __weak typeof(self) weakSelf = self;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
     RCTExecuteOnUIManagerQueue(^{
+      NSMutableArray *pendingUIBlocks = [self.uiManager valueForKey:@"_pendingUIBlocks"];
+      bool canPerformLayout = (pendingUIBlocks == nil) || ([pendingUIBlocks count] == 0);
+      
+      if (!canPerformLayout) {
+        dispatch_semaphore_signal(semaphore);
+      }
+      
       for (int i = 0; i < copiedOperationsQueue.count; i++) {
         copiedOperationsQueue[i](self.uiManager);
       }
-      [self.uiManager setNeedsLayout];
+      
+      if (canPerformLayout) {
+        weakSelf.shouldInterceptMountingBlock = YES;
```

Comment:
we already retain `self` (see, e.g. the line below), why adding another reference via weakSelf?

Reply:
I don't want to retain self because our block is used by UIManager so there would be a reference loop if UIManager was invalidated.